modules/solr/src/main/java/org/testcontainers/containers/SolrContainer.java

```diff
@@ -0,0 +1,130 @@
+package org.testcontainers.containers;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.net.URL;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.naming.ConfigurationException;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+
+import lombok.SneakyThrows;
+
+import org.apache.commons.lang.StringUtils;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+/**
+ * SolrContainer allows a solr container to be launched and controlled.
+ *
+ * @author Simon Schneider
+ */
+public class SolrContainer<SELF extends SolrContainer<SELF>> extends GenericContainer<SELF> {
+
+    public static final String IMAGE = "solr";
+    public static final String DEFAULT_TAG = "8.3.0";
+
+    public static final Integer ZOOKEEPER_PORT = 9983;
+    public static final Integer SOLR_PORT = 8983;
+
+    private SolrContainerConfiguration configuration;
+
+    public SolrContainer() {
+        this(IMAGE + ":" + DEFAULT_TAG);
+    }
+
+    public SolrContainer(final String dockerImageName) {
+        super(dockerImageName);
+        this.waitStrategy = new LogMessageWaitStrategy()
+            .withRegEx(".*o\\.e\\.j\\.s\\.Server Started.*")
+            .withStartupTimeout(Duration.of(60, SECONDS));
+        this.configuration = new SolrContainerConfiguration();
+    }
+
+    public SELF withZookeeper(boolean zookeeper) {
+        configuration.setZookeeper(zookeeper);
+        return self();
+    }
+
+    public SELF withCollection(String collection) {
+        if (StringUtils.isEmpty(collection)) {
+            throw new IllegalArgumentException();
+        }
```

Comment:
I think it would be helpful if the `IllegalArgumentException` had a human-friendly message.